nyxx/lib/src/core/channel/ThreadPreviewChannel.dart

```diff
@@ -0,0 +1,58 @@
+part of nyxx;
+
+/// Given when a thread is created as only partial information is available. If you want the final channel use [getThreadChannel]
+class ThreadPreviewChannel extends IChannel {
+  late final INyxx _client;
+  /// Name of the channel
+  late final String name;
+
+  /// Approximate message count
+  late final int messageCount;
+
+  /// Approximate member count
+  late final int memberCount;
+
+  /// Guild where the thread is located
+  late final Cacheable<Snowflake, Guild> guild;
+
+  /// The text channel where the thread was made
+  late final Cacheable<Snowflake, TextChannel> parentChannel;

```

Comment:
this could be `CacheableTextChannel` also

Reply:
Done